modules/common/src/main/scala/surge/internal/akka/actor/ActorLifecycleManagerActor.scala

```diff
@@ -3,34 +3,49 @@
 package surge.internal.akka.actor
 
 import akka.actor.{ Actor, ActorRef, Props, Terminated }
+import surge.internal.akka.actor.ActorLifecycleManagerActor.Ack
 
 object ActorLifecycleManagerActor {
   case object Start
   case object Stop
-
+  case class Ack(success: Boolean)
   def apply(managedActorProps: Props, managedActorName: Option[String]): ActorLifecycleManagerActor = {
     new ActorLifecycleManagerActor(managedActorProps, managedActorName)
   }
 }
 
-class ActorLifecycleManagerActor(managedActorProps: Props, managedActorName: Option[String] = None) extends Actor {
+class ActorLifecycleManagerActor(
+    managedActorProps: Props,
+    managedActorName: Option[String] = None,
+    initMessage: Option[() => Any] = None,
+    finalizeMessage: Option[() => Any] = None)
+    extends Actor {
   override def receive: Receive = stopped
 
   private def stopped: Receive = {
     case ActorLifecycleManagerActor.Start =>
       val actor = managedActorName match {
-        case Some(name) => context.actorOf(managedActorProps, name)
-        case _          => context.actorOf(managedActorProps)
+        case Some(name) =>
+          val actorRef = context.actorOf(managedActorProps, name)
+          initMessage.foreach(init => actorRef ! init())
```

Comment:
This line probably wants to be outside of the `match` - looks like currently if the actor isn't named it won't get the `init` message

Reply:
done